src/sql/workbench/parts/notebook/models/notebookModel.ts

```diff
@@ -744,8 +749,12 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		return newKernelDisplayName;
 	}
 
-	public addAttachToConnectionsToBeDisposed(connUri: string) {
+	public addAttachToConnectionsToBeDisposed(connUri: string, connectionProfile: ConnectionProfile) {
 		this._connectionUrisToDispose.push(connUri);
+
+		if (!this._connectionsForAttachTo.has(connUri)) {
```

Comment:
Is there any case where a a connection with a valid uri should be updated to a different connection profile?